scripts/zones/Port_Jeuno/npcs/Squintrox_Dryeyes.lua

```diff
@@ -65,53 +367,162 @@ entity.onTrigger = function(player, npc)
-    local lastViridian   = player:getCharVar('LastViridianKey') -- When last Viridian key was obtained
-
     if csid == 323 then
         if option == 1 then
+            -- catchall "stop wasting my time" response, triggered via updateEvent(1)
             player:showText(player, ID.text.DRYEYES_1)
-        elseif option == 100 then
-            if
-                salad and
-                juice and
-                grub and
-                acpMission >= xi.mission.id.acp.GATHERER_OF_LIGHT_I and
-                not hasViridianKey and
-                os.time() > lastViridian
-            then
-                player:addKeyItem(xi.ki.VIRIDIAN_KEY)
-                player:delKeyItem(xi.ki.BOWL_OF_BLAND_GOBLIN_SALAD)
-                player:delKeyItem(xi.ki.JUG_OF_GREASY_GOBLIN_JUICE)
-                player:delKeyItem(xi.ki.CHUNK_OF_SMOKED_GOBLIN_GRUB)
-                player:setCharVar('LastViridianKey', getMidnight())
-                player:showText(player, ID.text.DRYEYES_2)
-                player:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.VIRIDIAN_KEY)
-
-            elseif
-                os.time() <= lastViridian or
-                hasViridianKey
-            then
-                player:messageSpecial(ID.text.DRYEYES_3, xi.ki.VIRIDIAN_KEY)
-
+        elseif option == 100 then -- Viridian Key
+            takeReqKeyItems(player, xi.ki.VIRIDIAN_KEY)
+        elseif option == 101 then -- blue coral Key
+            takeReqKeyItems(player, xi.ki.BLUE_CORAL_KEY)
+        elseif option == 102 then -- black coral Key
+            -- shouldn't trigger without the moldy chest, but just in case since the logic above is complex
+            local ki = xi.ki.BLACK_CORAL_KEY
+            if player:hasKeyItem(menuMetadata[1][ki].reqKeyItems[1]) then
+                takeReqKeyItems(player, ki)
+            end
+        elseif option == 103 then -- Bird Key prereqs to run the bcnms
+            npcUtil.giveKeyItem(player, menuMetadata[1][xi.ki.BIRD_KEY].prereqKeyItems)
+        elseif option == 104 then -- Bomb Key
+            takeReqKeyItems(player, xi.ki.BOMB_KEY)
+        elseif option == 105 then -- Bird Key
+            takeReqKeyItems(player, xi.ki.BIRD_KEY)
+        elseif option >= 300 and option <= 308 then
+            local entry = menuMetadata[2][option - 299]
+            if player:getSeals(0) < entry.costSeals then
+                player:messageSpecial(ID.text.DRYEYES_4)
+            elseif player:getGil() < entry.costGil then
+                player:messageSpecial(ID.text.DRYEYES_5)
             else
-                -- player:showText(player, ?)
-                -- Doesn't seem to be a message for trying when you don't have the key items?
+                player:delSeals(entry.costSeals, 0)
+                player:delGil(entry.costGil)
+
+                for _, ki in pairs(entry.relevantKeyItems) do
+                    if not player:hasKeyItem(ki) then
+                        player:addKeyItem(ki)
```

Comment:
These two lines can be replaced by the npcUtil function (which can handle either single input or tables of KIs)

Reply:
Good catch.